core/src/kura.rs

```diff
@@ -278,7 +353,7 @@ impl Kura {
 /// The [`BlockStoreTrait`] defines and implements functionality used by Kura
 /// for every platform. The default implementation is `StdFileBlockStore`,
 /// which uses `std::fs`.
-pub trait BlockStoreTrait {
+pub trait BlockStoreTrait : Debug {
```

Comment:
Technically this doesn't hurt, but I find that this anti-pattern of requiring `Debug` is only needed when you want to print something with a trai-bound `T: BlockStoreTrait`. 

In general this trait should go, given that we don't really have any implementation other than Kura, and rely on the actor system to the extent to which alternative implementations would be difficult.

Reply:
It's because other objects have derive. They need a reference to the Kura subsystem so they need an Arc<Kura>. This requires Kura to also implement debug and therefore BlockStoreTrait as well. We should probably do a custom implementation that doesn't print anything.